File: Dynamic Programming Applications In Machine Learning and Genomics/Longest Path in a DAG/LongestPathDAG.java
import java.util.Scanner;

class LongestPathDAG {
    /**
     * @param source is the integer label of the source node
     * @param sink is the integer label of the sink node
     * @param edges is a String array of the edges as represented in the input file
     *
     * @return a string containing the output in the correct format
     */
    private static String longestPath(int source, int sink, String[] edges) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int source = stdin.nextInt();
        int sink = stdin.nextInt();

        stdin.nextLine();
        
        stdin.useDelimiter("\\Z");
        String[] edges = stdin.next().split("\\n");

        System.out.println(longestPath(source, sink, edges));
    }
}

File: Dynamic Programming Applications In Machine Learning and Genomics/Longest Path in a DAG/longest_path_dag.py
#python3
import sys

def longest_path(source, sink, edges):
    """
    source is the integer source node label
    sink is the integer sink node lable
    edges is a string list of the edges from the input file
    
    return a string containing the output in the correct format
    """
    # TODO: your code here
    return ""

if __name__ == "__main__":
    source = int(sys.stdin.readline().strip())
    sink = int(sys.stdin.readline().strip())
    edges = [line.strip() for line in sys.stdin]

    print(longest_path(source,sink,edges))
